### ISSUE TITLE
Where are the benchmarks

### ISSUE THREAD

**Issue by Eilie** [jorisvink/kore#57]
Yet another C web server but without benchmarks - NO WAY!

**Comment by apfohl** [comment]
I'm working on some benchmarks.

**Comment by jorisvink** [comment]
@Eilie I'm not sure opening issues like this is a productive approach. It just shows immaturity and honestly a serious lack of respect towards open source projects.

Having said that, I personally dislike publishing benchmark data. There are literally a dozen ways of creating benchmarks which show at most a base line of what something is capable of. While having a base line is really good it serves no purpose in most cases where the bottlenecks are almost always going to lie in your application code.

If people who know what they are doing want to create sensible benchmarks and publish those results I am all game for it.

TL;DR but want numbers? Ok fine I'll bite.

On my workstation (Xeon E3-1245 @ 3.40GHz, 8 cores) I am able to throw over 600.000 req/sec for doing something as stupid as serving a 100-byte file (note that this not over TLS as we want to measure Kore rather then the TLS library it is linked again). That is with 4 worker processes and 4 benchmark processes running with 1000 concurrent connections.

That kind of performance is going to be more then enough for most people wanting to use Kore for something public facing. There is only a handful of scenarios where you are going to pull more HTTP requests / second and those are usually reserved for only a certain type of companies.

TL;DR 2 - What is the most connections ever thrown at Kore? 1.8 million concurrent.

Is that type of information useful? Maybe, maybe not.

Feel free to give it a spin and judge for yourself if its something worth using on your projects or not.

**Comment by Eilie** [comment]
@jorisvink Yeah, I've tried to launch example myself but failed due to need to install it system-wide with `make install` which I don't want(running arbitrary code from internet is not a good idea let alone installing it in system, and I don't have time for auditing). Would be cool to have it work in standalone fashion somehow. It builds `kore` binary but I'm not smart enough to figure out how to use it to run example application on my own, probably extended README would help.
Tried:
```bash
git clone https://github.com/jorisvink/kore.git
cd kore
make -j9
./kore create hello
./kore run hello
compiling hello.c
hello/src/hello.c:1:23: fatal error: kore/kore.h: No such file or directory
 #include <kore/kore.h>
                       ^
compilation terminated.
kore run: subprocess trouble, check output
```

**Comment by Eilie** [comment]
@jorisvink Just a quick tip, some benchmark comparison charts + environment info would help, look at 
 https://h2o.github.io/, http://lwan.ws/, etc for inspiration, that's how new HTTP servers roll today(and ever). Comparison with old and popular HTTP servers is important because it'll help to figure out how much server money one will save by switching to something new and fast.

**Comment by jorisvink** [comment]
You could always use a VM for code you really don't trust.

The examples assume Kore is installed on your system because they use the built in commands to build them. If you want you can of course link together your own dynamic library, build a config file and run Kore from whichever local directory you want.

Baseline charts are sometimes useful yes, but you'll have to produce them yourself to get a good idea of what performance level you are looking for.

**Comment by jorisvink** [comment]
Closing this issue as its been open > 1 month with no real resolution or additional input.
